src/NCronJob/Configuration/Builder/JobOptionBuilder.cs

```diff
@@ -30,6 +30,24 @@ public ParameterBuilder WithCronExpression(string cronExpression, TimeZoneInfo?
         return new ParameterBuilder(this, jobOption);
     }
 
+    /// <summary>
+    /// Sets the job name. This can be used to identify the job.
+    /// </summary>
+    /// <param name="jobName">The job name associated with this job.</param>
+    /// <returns>Returns a <see cref="ParameterBuilder"/> that allows further configuration.</returns>
+    /// <remarks>The job name should be unique over all job instances.</remarks>
+    public ParameterBuilder WithName(string jobName)
```

Comment:
Hmm now that isn't the best choice. The user can do the following:
```csharp
j => j.WithName("Foo").WithName("Bar");
```

As the `ParameterBuilder` also has a `ParameterBuilder`.

To remove the ambiguity, we could have a `WithParameter` property that directly returns the `ParameterBuilder`:
```csharp
j => j.WithParameter.WithName("Foo");
```

Reply:
On it.

Reply:
I came up with a different proposal. Thoughts?

- Prevents the WithName().WithName() issue
- Still allows the WithParameter().WithParameter() usage
- Backward compatible (from an API standpoint)

May need to be cleaned up in v5 to come up with something cleaner.